crates/iota-framework/packages/iota-system/sources/iota_system_state_inner.move

```diff
@@ -821,18 +881,19 @@ module iota_system::iota_system_state_inner {
             self.parameters.validator_low_stake_threshold,
             self.parameters.validator_very_low_stake_threshold,
             self.parameters.validator_low_stake_grace_period,
+            self.parameters.committee_members_count,
             ctx,
         );
 
-        let new_total_stake = self.validators.total_stake();
+        let new_total_stake = self.validators.total_stake_inner();
 
         let remaining_validator_rewards_amount_after_distribution = total_validator_rewards.value();
         let total_stake_rewards_distributed = total_validator_rewards_amount_before_distribution - remaining_validator_rewards_amount_after_distribution;
 
         self.protocol_version = next_protocol_version;
 
         // Derive the reference gas price for the new epoch
-        self.reference_gas_price = self.validators.derive_reference_gas_price();
+        self.reference_gas_price = self.validators.derive_reference_gas_price_inner();
```

Comment:
should we clean up all the gas price survey with this change?

Reply:
Was there a reason this wasn't removed with PCR2? Maybe we should do it with protocol-version-5, but definitely on a separate PR.

Reply:
remove this line

Reply:
removed